ansible/roles/test/files/ptftests/vxlan_traffic.py

```diff
@@ -0,0 +1,271 @@
+# ptf --test-dir ptftests vxlan_traffic.VXLAN --platform-dir ptftests --qlen=1000 --platform remote \
+#    -t 't2_ports=[16, 17, 0, 1, 4, 5, 21, 20];dut_mac=u"64:3a:ea:c1:73:f8";expect_encap_success=True; \
+#        vxlan_port=4789;topo_file="/tmp/vxlan_topo_file.json";config_file="/tmp/vxlan-config-TC1-v6_in_v4.json";t0_ports=[u"Ethernet42"]' --relax --debug info \
+#        --log-file /tmp/vxlan-tests.TC1.v6_in_v4.log
+
+# The test checks vxlan encapsulation:
+# The test runs three tests for each vlan on the DUT:
+# 'test_encap' : Sends regular packets to T0-facing interface and expects to see the encapsulated packets on the T2-facing interfaces.
+#
+# The test has the following parameters:
+# 1. 'config_file' is a filename of a file which contains all necessary information to run the test. The file is populated by ansible. This parameter is mandatory.
+
+import sys
+import os.path
+import json
+import ptf
+import ptf.packet as scapy
+from ptf.base_tests import BaseTest
+from ptf import config
+from ptf.testutils import *
+from ptf.dataplane import match_exp_pkt
+from ptf.mask import Mask
+import datetime
+import subprocess
+import ipaddress
+from pprint import pprint
+from ipaddress import ip_address
+import random
+VARS = {}
+VARS['tcp_sport'] = 1234
+VARS['tcp_dport'] = 5000
+
+# Some constants used in this code
+TEST_ECN = False
+
+def get_incremental_value(key):
+
+    global VARS
+    VARS[key] = VARS[key] + 1
+    return VARS[key]
+
+def read_ptf_macs():
+    addrs = {}
+    for intf in os.listdir('/sys/class/net'):
+        if os.path.isdir('/sys/class/net/%s' % intf):
+            with open('/sys/class/net/%s/address' % intf) as fp:
+                addrs[intf] = fp.read().strip()
+
+    return addrs
+
+class VXLAN(BaseTest):
+    def __init__(self):
+        BaseTest.__init__(self)
+        self.DEFAULT_PKT_LEN = 100
+
+    def setUp(self):
+        self.dataplane = ptf.dataplane_instance
+        self.test_params = test_params_get()
+        self.dut_mac = self.test_params['dut_mac']
+        self.vxlan_port = self.test_params['vxlan_port']
+        self.expect_encap_success = self.test_params['expect_encap_success']
+
+        self.random_mac = "00:aa:bb:cc:dd:ee"
+        self.ptf_mac_addrs = read_ptf_macs()
+        with open(self.test_params['config_file']) as fp:
+            self.config_data = json.load(fp)
+        with open(self.test_params['topo_file']) as fp:
+            self.topo_data = json.load(fp)
+
+        self.fill_loopback_ip()
+        self.routes_removed = self.test_params.get('routes_removed', False)
```

Comment:
Is there anywhere in the test case that used this? I think it should be a scenario but could not find it anywhere.

Reply:
I will remove it.

Reply:
I thought the scenario of the routes being removed should be tested as well. So instead of removal, should we do something in the test cases to cover this condition?

Reply:
Yes, It is already covered in test cases listed in the HLD.